tests/integration/coordinator/plugins/clickhouse/test_zookeeper.py

```diff
@@ -125,3 +131,108 @@ async def test_kazoo_zookeeper_client_bounded_failure_time(ports: Ports) -> None
+        async with zookeeper_client.connect() as connection:
+            await connection.create("/test_fails_if_node_is_deleted/child1", b"child1")
+            await connection.create("/test_fails_if_node_is_deleted/child2", b"child2")
+            await connection.create("/test_fails_if_node_is_deleted/child3", b"child3")
+            task = asyncio.create_task(
+                connection.get_children_with_data("/test_fails_if_node_is_deleted", inject_fault=self.get_sleep_always())
+            )
+            await connection.delete("/test_fails_if_node_is_deleted", recursive=True)
+
+            with pytest.raises(NoNodeError):
+                await task
+
+    async def test_works_with_adding_node(self, zookeeper_client: KazooZooKeeperClient) -> None:
+        async with zookeeper_client.connect() as connection:
+            await connection.create("/test_works_with_adding_node/child1", b"child1")
+            await connection.create("/test_works_with_adding_node/child2", b"child2")
+            await connection.create("/test_works_with_adding_node/child3", b"child3")
+            task = asyncio.create_task(
+                connection.get_children_with_data("/test_works_with_adding_node", inject_fault=self.get_sleep_once())
+            )
+            await connection.create("/test_works_with_adding_node/child4", b"child4")
+            assert (await task) == {"child1": b"child1", "child2": b"child2", "child3": b"child3", "child4": b"child4"}
+
+    async def test_works_with_removing_node(self, zookeeper_client: KazooZooKeeperClient) -> None:
+        async with zookeeper_client.connect() as connection:
+            await connection.create("/test_works_with_removing_node/child1", b"child1")
+            await connection.create("/test_works_with_removing_node/child2", b"child2")
+            await connection.create("/test_works_with_removing_node/child3", b"child3")
+            task = asyncio.create_task(
+                connection.get_children_with_data("/test_works_with_removing_node", inject_fault=self.get_sleep_once())
+            )
+            await connection.delete("/test_works_with_removing_node/child2")
+            assert (await task) == {"child1": b"child1", "child3": b"child3"}
+
+    async def test_works_with_changing_node(self, zookeeper_client: KazooZooKeeperClient) -> None:
+        async with zookeeper_client.connect() as connection:
+            await connection.create("/test_works_with_changing_node/child1", b"child1")
+            await connection.create("/test_works_with_changing_node/child2", b"child2")
+            await connection.create("/test_works_with_changing_node/child3", b"child3")
+            task = asyncio.create_task(
+                connection.get_children_with_data("/test_works_with_changing_node", inject_fault=self.get_sleep_once())
+            )
+            await connection.set("/test_works_with_changing_node/child2", b"child2_changed")
```

Comment:
I do not understand how those processes are synchronized.
Looks like `set` may happen before task starts.
Could it be possible to call `set` in the `inject_fault` callback?